core/variant/variant_parser.h

```diff
@@ -47,6 +47,7 @@ class VariantParser {
 
 	protected:
 		virtual uint32_t _read_buffer(char32_t *p_buffer, uint32_t p_num_chars) = 0;
+		uint32_t _get_buffer_remaining_length() const { return readahead_filled - readahead_pointer; }
 
```

Comment:
Note that when EOF is reached in `VariantParser::Stream::get_char()`, `readahead_pointer` is set to 1 and `readhead_filled` is zero, in order to catch bugs etc. So this will return `UINT32_MAX`.